### ISSUE TITLE
Staking: allow nomination mirroring

### ISSUE THREAD

**Issue by kianenigma** [paritytech/substrate#13862]
A simple solution to allow application such as https://forum.polkadot.network/t/nomination-pool-controlled-by-a-smart-contract/2436/2 would be to tweak `Nomination::targets` to be, instead of a plain `Vec<AccountId>`:

```
enum NominationTarget {
    /// Directly nominating thyself. 
    Direct(Vec<AccountId>),
    /// Mirroring the nomination preference of another `AccountId`.
    Mirror(AccountId)
} 
```

This will possible increase the weight of the election snapshot, but given that we already read most nominators' data anyway, it is mostly going to be overlay overhead, not actual IO read. 

Chaining should probably not be allowed. 

If you are `Mirror(Alice)` and `Alice` is anything but `Direct(_)`, then you are not taken into account and are basically in an "inactive" state.

This sounds not too hard, and enables a bunch of things, but going back to the original issue, the real solution is keep direct nominations raw and simple, and build such functionalities into pools.

**Comment by Polkadot-Forum** [comment]
This issue has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/nomination-pool-controlled-by-a-smart-contract/2436/4

**Comment by bkchr** [comment]
And what triggers the smart contract?

**Comment by rphmeier** [comment]
Either an on-chain scheduler or a system where anyone can post a transaction to trigger the update, paying the (small) gas for that.

**Comment by kianenigma** [comment]
> Either an on-chain scheduler or a system where anyone can post a transaction to trigger the update, paying the (small) gas for that.

Indeed, but this brings up the topic of smart-contracts on the relay chain? We discussed this internally and concluded to not pursue it any further. Once staking lives in its own parachain, something like this can easily be done via contracts.
